include/dlaf/factorization/qr/t_factor_impl.h

```diff
@@ -324,7 +324,7 @@ void QR_Tfactor<backend, device, T>::call(
 
   // at this point each rank has its partial result for each column
   // so, let's reduce the results (on all ranks, so that everyone can independently compute T factor)
-  if (mpi_col_task_chain.size_2d().rows() > 0)
+  if (mpi_col_task_chain.size_2d().rows() > 0 && mpi_col_task_chain.size() > 1)
```

Comment:
We already attempted at fixing this, but I failed 😅  https://github.com/eth-cscs/DLA-Future/pull/993#discussion_r1397050103

My new suggestion is
```suggestion
  if (mpi_col_task_chain.size() > 1)
```